src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/WebClientConfiguration.kt

```diff
@@ -1,73 +1,85 @@
+import org.springframework.web.reactive.function.client.ExchangeFunction
 import org.springframework.web.reactive.function.client.WebClient
 import reactor.netty.http.client.HttpClient
 import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.SYSTEM_USERNAME
-import java.time.Duration
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.utils.UserContext
+import java.util.*
 
 @Configuration
 class WebClientConfiguration(
-  @Value("\${api.base.url.oauth}") val authBaseUri: String,
+  @Value("\${api.base.url.oauth}") private val authBaseUri: String,
   @Value("\${api.base.url.prison}") private val prisonRootUri: String,
 ) {
 
   @Bean
-  fun authWebClient(): WebClient {
-    return WebClient.builder()
+  fun webClient(builder: WebClient.Builder): WebClient {
+    return builder
       .baseUrl(authBaseUri)
+      .filter(addAuthHeaderFilterFunction())
       .build()
   }
 
   @Bean
-  fun prisonWebClient(): WebClient {
-    val httpClient = HttpClient.create().responseTimeout(Duration.ofMinutes(2))
-    return WebClient.builder()
-      .baseUrl(prisonRootUri)
-      .clientConnector(ReactorClientHttpConnector(httpClient))
-      .filter(AuthTokenFilterFunction())
-      .build()
-  }
+  fun prisonWebClientClientCredentials(
+    @Qualifier(value = "authorizedClientManagerAppScope") authorizedClientManager: OAuth2AuthorizedClientManager,
+    builder: WebClient.Builder,
+  ): WebClient = getOAuthWebClient(authorizedClientManager, builder, prisonRootUri)
 
   @Bean
-  fun prisonWebClientClientCredentials(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager): WebClient {
-    val oauth2Client = ServerOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
-    oauth2Client.setDefaultClientRegistrationId(SYSTEM_USERNAME)
+  fun prisonWebClient(builder: WebClient.Builder): WebClient = createForwardAuthWebClient(builder, prisonRootUri)
 
-    val httpClient = HttpClient.create().responseTimeout(Duration.ofMinutes(2))
-    return WebClient.builder()
-      .baseUrl(prisonRootUri)
-      .clientConnector(ReactorClientHttpConnector(httpClient))
-      .filter(oauth2Client)
+  private fun getOAuthWebClient(
+    authorizedClientManager: OAuth2AuthorizedClientManager,
+    builder: WebClient.Builder,
+    rootUri: String,
+  ): WebClient {
+    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    oauth2Client.setDefaultClientRegistrationId(SYSTEM_USERNAME)
+    return builder.baseUrl(rootUri)
+      .apply(oauth2Client.oauth2Configuration())
       .build()
   }
 
-  @Bean
-  fun prisonHealthWebClient(): WebClient {
-    return WebClient.builder()
-      .baseUrl(prisonRootUri)
-      .build()
+  private fun createForwardAuthWebClient(builder: WebClient.Builder, url: @URL String): WebClient = builder.baseUrl(url)
+    .filter(addAuthHeaderFilterFunction())
+    .clientConnector(
+      ReactorClientHttpConnector(HttpClient.create()),
```

Comment:
normally people add a `clientConnector` to set ` responseTimeout` on the `HttpClient`, otherwise I think the default is the same as what you've provided here anyway